README.md

```diff
@@ -1,14 +1,126 @@
+## General
+
+Oracle connector is one of [Conduit](https://github.com/ConduitIO/conduit) plugins. It provides both, a source and
+destination Oracle connector.
+
+## Prerequisites
 
-### Prerequisites
 - [Go](https://go.dev/) 1.18
 - (optional) [golangci-lint](https://github.com/golangci/golangci-lint) 1.49.0
+- (optional) [mock](https://github.com/golang/mock) 1.6.0
+
+## How to build it
 
-### How to build it
 Run `make build`.
 
-### Testing
-Run `make test`.
\ No newline at end of file
+## Testing
+
+Run `make test` to run all unit and integration tests, as well as an acceptance test. To make integration and acceptance
+tests pass, set the Oracle database URL to the environment variables as an `ORACLE_URL`.
+
+## Source
+
+The Oracle Source connects to the database using the provided `url` and starts creating records for each table row and
+each change detected. The first time Source runs, it creates an additional table and a trigger to track changes in
+the `table` and launches Snapshot mode. Then, when all the records have been read, Source switches to CDC mode. More
+information [inside the Change Data Capture section](#change-data-capture).
+
+### Snapshot
+
+The connector in the Snapshot mode reads all rows from the table in batches via SELECT, FETCH NEXT and ORDER BY statements.
+
+Example of a query:
+
+```
+SELECT {{columns...}}
+FROM {{table}}
+ORDER BY {{orderingColumn}}
+WHERE {{keyColumn}} > {{position.last_processed_val}}
+FETCH NEXT {{batchSize}} ROWS ONLY;
+```
+
+When all records have been returned, the connector switches to the CDC mode.
+
+### Change Data Capture
+
+This connector implements CDC features for Oracle by adding a tracking table and a trigger to populate it. The tracking
+table and trigger name have the same names as a target table wit the prefix `CONDUIT`. The tracking table has all the
+same columns as the target table plus three additional columns:
+
+| name                          | description                                          |
+|-------------------------------|------------------------------------------------------|
+| `CONDUIT_TRACKING_ID`         | Autoincrement index for the position.                |
+| `CONDUIT_OPERATION_TYPE`      | Operation type: `insert`, `update`, or `delete`.     |
+| `CONDUIT_TRACKING_CREATED_AT` | Date when the event was added to the tacking table.  |
+
+Every time data is added, changed, or deleted from the target table, this event will be written to the tracking table.
+
+The queries to get change data from the tracking table look pretty similar to queries in the Snapshot mode, but
+with `CONDUIT_TRACKING_ID` as an ordering column.
+
+The Ack method collects the `CONDUIT_TRACKING_ID` of those records that have been successfully applied, in order to
+remove them later in a batch from the tracking table (every 5 seconds or when the connector is closed).
+
```

Comment:
please add a few words about what will be if iterator stops the run, How it will use position